pkg/destroy/aws/aws.go

```diff
@@ -77,1510 +63,1205 @@ func (o *ClusterUninstaller) validate() error {
 	return nil
 }
 
-// populateDeleteFuncs is the list of functions that will be launched as goroutines
-func populateDeleteFuncs(funcs map[string]deleteFunc) {
-	funcs["deleteVPCs"] = deleteVPCs
-	funcs["deleteEIPs"] = deleteEIPs
-	funcs["deleteNATGateways"] = deleteNATGateways
-	funcs["deleteInstances"] = deleteInstances
-	funcs["deleteIAMresources"] = deleteIAMresources
-	funcs["deleteSecurityGroups"] = deleteSecurityGroups
-	funcs["deleteInternetGateways"] = deleteInternetGateways
-	funcs["deleteSubnets"] = deleteSubnets
-	funcs["deleteS3Buckets"] = deleteS3Buckets
-	funcs["deleteRoute53"] = deleteRoute53
-	funcs["deletePVs"] = deletePVs
-	funcs["deleteUsers"] = deleteUsers
-}
-
 // Run is the entrypoint to start the uninstall process
 func (o *ClusterUninstaller) Run() error {
 	err := o.validate()
 	if err != nil {
 		return err
 	}
 
-	deleteFuncs := map[string]deleteFunc{}
-	populateDeleteFuncs(deleteFuncs)
-	returnChannel := make(chan string)
+	awsConfig := &aws.Config{Region: aws.String(o.Region)}
 
-	awsSession, err := getAWSSession(o.Region)
+	// Relying on appropriate AWS ENV vars (eg AWS_PROFILE, AWS_ACCESS_KEY_ID, etc)
+	awsSession, err := session.NewSession(awsConfig)
 	if err != nil {
 		return err
 	}
 
-	// launch goroutines
-	goroutines := 0
-	for name, function := range deleteFuncs {
-		for _, filter := range o.Filters {
-			go deleteRunner(name, function, awsSession, filter, o.ClusterName, o.Logger, returnChannel)
-			goroutines++
+	tagClients := []*resourcegroupstaggingapi.ResourceGroupsTaggingAPI{
+		resourcegroupstaggingapi.New(awsSession),
+	}
+	tagClientNames := map[*resourcegroupstaggingapi.ResourceGroupsTaggingAPI]string{
+		tagClients[0]: o.Region,
+	}
+	if o.Region != "us-east-1" {
```

Comment:
Hmm, why this?

Reply:
> I'll work up a reproducer and add it to the commit message.

Done with e7b4c4e -> b44da20 (notes in 3c5cef475).